smart-contract-verifier/smart-contract-verifier-server/src/types/lookup_methods.rs

```diff
@@ -0,0 +1,56 @@
+use crate::proto;
+use amplify::{From, Wrapper};
+use blockscout_display_bytes::Bytes as DisplayBytes;
+use ethers_core::abi::Abi;
+use ethers_solc::sourcemap;
+use smart_contract_verifier::{LookupMethodsRequest, LookupMethodsResponse};
+use std::{collections::BTreeMap, str::FromStr};
+
+#[derive(Wrapper, From, Clone, Debug, PartialEq)]
+pub struct LookupMethodsRequestWrapper(proto::LookupMethodsRequest);
+
+impl TryFrom<LookupMethodsRequestWrapper> for LookupMethodsRequest {
+    type Error = tonic::Status;
+
+    fn try_from(request: LookupMethodsRequestWrapper) -> Result<Self, Self::Error> {
+        let bytecode = DisplayBytes::from_str(&request.bytecode)
+            .map_err(|e| tonic::Status::invalid_argument(format!("Invalid bytecode: {e:?}")))?
+            .0;
+        let abi = Abi::load(request.abi.as_bytes())
+            .map_err(|e| tonic::Status::invalid_argument(format!("Invalid abi: {e:?}")))?;
+        let source_map = sourcemap::parse(&request.source_map)
+            .map_err(|e| tonic::Status::invalid_argument(format!("Invalid source_map: {e:?}")))?;
+        let file_ids = request.file_ids.clone();
+
+        Ok(Self {
+            bytecode,
+            abi,
+            source_map,
+            file_ids,
+        })
+    }
+}
+
+#[derive(Wrapper, From, Clone, Debug, PartialEq)]
+pub struct LookupMethodsResponseWrapper(proto::LookupMethodsResponse);
+
+impl From<LookupMethodsResponse> for LookupMethodsResponseWrapper {
+    fn from(response: LookupMethodsResponse) -> Self {
+        Self(proto::LookupMethodsResponse {
+            methods: response
+                .methods
+                .iter()
```

Comment:
Nit: allows to get rid of clones in the map
```suggestion
                .into_iter()
```